yapio.c

```diff
@@ -2135,6 +2180,37 @@ yapio_display_result(const yapio_test_ctx_t *ytc)
     }
 }
 
+/* dump yapioMyRank's metadata array into a file */
+static void
+yapio_store_md_final_state()
+{
+    /* each rank writes its own md file '{filename}-md-{rank}' */
+    char md_file[PATH_MAX + 10];
+    snprintf(md_file, PATH_MAX, "%s-md-%d", yapioTestFileName, yapioMyRank);
+
+    yapio_test_group_t *ytg = yapioMyTestGroup;
+    const size_t nblks_per_rank = ytg->ytg_num_blks_per_rank;
+
+    /* use metadata from last iteration */
+    int test_ctx_idf = ytg->ytg_num_contexts - 1;
+    yapio_test_ctx_t *ytc = &ytg->ytg_contexts[test_ctx_idf];
+
+    FILE *file = fopen(md_file, "w+");
+
+    if (file)
+    {
+        int num_ops_in = 0;
+
+        const yapio_blk_md_t *md_array =
+            yapio_test_ctx_to_md_array(ytc, YAPIO_TEST_CTX_MDH_OUT, &num_ops_in);
+
+        fwrite(md_array, sizeof(yapio_blk_md_t), nblks_per_rank, file);
+        fclose(file);
+    }
+    else
+        printf("Unable to open file\n");
+}
```

Comment:
no "yapio_print_help" to exit in error ?

Reply:
adding it